### PR TITLE
HWP-469: Backend Search

### PR DESCRIPTION
**Pull request by BradyMitch**
# Description

This PR includes the following proposed change(s):

- Added **GET /api/search/{query}**
- Search query is case insensitive
- Returns `{ users: [], communities: [], posts: [] }`
- If no results match your search, returns **"No results found."**
- Search users by username, firstName, lastName, or fullName.
- Search communities by title.
- Search posts by title and message.
- Advanced search query `author:<username>` or `creator:<username>` returns posts created by <username>. Use like: `author:jeff test`
- Advanced search query `latestActivity: ` returns communities in order of latestActivity. Use like `latestActivity: test`

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Tested searching for users, posts, and communities in swagger.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [x] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [x] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/express.js [hunk 1]
```diff
@@ -137,6 +138,8 @@ app.use("/api/login", initLogger, loginRouter, requestFinally);
 app.use("/api/logout", initLogger, logoutRouter, requestFinally);
 app.use("/api/token", initLogger, tokenRouter, requestFinally);
 
+app.use("/api/search", initLogger, searchRouter, requestFinally);
```
Comment (ZachBourque) [inline app/server/express.js:141]: I may be overlooking something, but what's the reasoning for not calling authenticateToken on this endpoint?

Comment (BradyMitch) [inline app/server/express.js:141]:
```suggestion
app.use("/api/search", initLogger, authenticateToken, searchRouter, requestFinally);
```

Comment (BradyMitch) [reply app/server/express.js]: Thanks, that should have been there.

**Review by ZachBourque — APPROVED** [review]
Works well!

**Pull request merged**